packages/vm/src/imports.rs

```diff
@@ -537,6 +545,60 @@ mod test {
         assert_eq!(result, errors::REGION_READ_LENGTH_TOO_BIG);
     }
 
+    #[test]
+    fn do_remove_works() {
+        let mut instance = make_instance();
+
+        let existing_key = KEY1;
+        let key_ptr = write_data(&mut instance, existing_key);
+
+        let ctx = instance.context_mut();
+        leave_default_data(ctx);
+
+        let result = do_remove::<S, Q>(ctx, key_ptr);
+        assert_eq!(result, errors::NONE);
+
+        let value = with_storage_from_context::<S, Q, _, _>(ctx, |store| {
+            Ok(store.get(existing_key).expect("error getting value"))
```

Comment:
Can't you just do `|store| store.get(existing_key)`?
Rather than unwrapping it and casting success as okay inside the closure. We unwrap the result on the line below.

Reply:
No so easy, since `with_storage_from_context` returns `VmResult` and `store.get` returns `StdResult`. I unwrap it directly to make it easier to understand the two unwraps.

Reply:
Ah. Good point.

We definitely do need to revisit these error types. At least they have unique names now.